linera-execution/src/execution_state_actor.rs

```diff
@@ -228,11 +225,25 @@ where
                     epoch: self.system.epoch.get().ok_or_else(inactive_err)?,
                     committees: self.system.committees.get().clone(),
                     balance,
-                    authorized_applications,
+                    application_permissions,
                 };
                 let messages = self.system.open_chain(config, next_message_id)?;
                 callback.respond(messages)
             }
+
+            CloseChain {
+                application_id,
+                callback,
+            } => {
+                let app_permissions = self.system.application_permissions.get();
+                if !app_permissions.can_close_chain(&application_id) {
+                    callback.respond(Err(ExecutionError::UnauthorizedApplication(application_id)));
```

Comment:
Yes, we could make `open_chain` return an `Option<ChainId>`… or introduce an error type for it: It can either fail because the app isn't authorized, or because we have insufficient balance.

I didn't do it for now to be consistent with `transfer`; but it would probably better if all of them returned results.

Reply:
I made it return a `bool` for now. Trying to return a result I'm running into wasm-bindgen issues. Maybe I should do that in a separate PR, and then also for `transfer`?

Reply:
A bool sounds great.